File: 2026-2/sesion-de-clase-09/softprog/softprog-modelo/src/main/java/pe/edu/pucp/softprog/modelo/seguridad/CuentaUsuario.java
package pe.edu.pucp.softprog.modelo.seguridad;

import pe.edu.pucp.softprog.modelo.Registro;


public class CuentaUsuario extends Registro {
    private String userName;
    private String password ;

    public CuentaUsuario() {
    }

    public CuentaUsuario(final CuentaUsuario cuentaUsuario) {
        if  (cuentaUsuario == null) {
            throw new IllegalArgumentException("cuentaUsuario no puede ser nulo");
        }
        super(cuentaUsuario);
        setUserName(cuentaUsuario.getUserName());
        setPassword(cuentaUsuario.getPassword());
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        if (userName == null || userName.isEmpty()) {
            throw new IllegalArgumentException("userName no puede ser nulo o vacío");
        }

        this.userName = userName;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        if (password == null || password.isEmpty()) {
            throw new IllegalArgumentException("password no puede ser nulo o vacío");
        }

        this.password = password;
    }
}

File: 2026-2/sesion-de-clase-11/SoftProg/SoftProg.Modelo/Seguridad/CuentaUsuario.cs
namespace SoftProg.Modelo.Seguridad {
    public class CuentaUsuario : Registro {
        public CuentaUsuario() { }

        public CuentaUsuario(CuentaUsuario cuentaUsuario) : base(cuentaUsuario) {
            UserName = cuentaUsuario.UserName;
            Password = cuentaUsuario.Password;
        }

        public string UserName { 
            get;
            set { 
                ArgumentNullException.ThrowIfNullOrEmpty(value);
                field = value;
            }
        }
        public string Password { 
            get;
            set {
                ArgumentNullException.ThrowIfNullOrEmpty(value);
                field = value;
            }
        }
    }
}
